src/test/test_ipaddr.cc

```diff
@@ -995,3 +995,172 @@ TEST(pick_address, ipv4_ipv6_enabled2)
     ASSERT_EQ(-1, r);
   }
 }
+
+void setup_ipv4(struct sockaddr_in* addr, const std::string& ip) {
+  memset(addr, 0, sizeof(*addr));
+  addr->sin_family = AF_INET;
+  inet_pton(AF_INET, ip.c_str(), &addr->sin_addr);
+}
+
+// Helper function to set up IPv6 sockaddr_in6
+void setup_ipv6(struct sockaddr_in6* addr, const std::string& ip) {
+  memset(addr, 0, sizeof(*addr));
+  addr->sin6_family = AF_INET6;
+  inet_pton(AF_INET6, ip.c_str(), &addr->sin6_addr);
+}
+
+// Test for IPv4 address
+TEST(is_addr_in_subnet, ipv4)
+{
+  struct sockaddr_in a_one;
+
+  setup_ipv4(&a_one, "10.1.1.2");
```

Comment:
Is `a_one` ever used after the setup? If not, what's the benefit of it?

Reply:
no, redundant, removing